internal/datanode/data_node.go

```diff
@@ -367,28 +373,51 @@ func parseDeleteEventKey(key string) string {
 	return vChanName
 }
 
-func (node *DataNode) handlePutEvent(watchInfo *datapb.ChannelWatchInfo) error {
+func (node *DataNode) handlePutEvent(watchInfo *datapb.ChannelWatchInfo, version int64) (err error) {
 	vChanName := watchInfo.GetVchan().GetChannelName()
 
-	if err := node.flowgraphManager.addAndStart(node, watchInfo.GetVchan()); err != nil {
-		return fmt.Errorf("fail to add and start flowgraph for vChanName: %s, err: %v", vChanName, err)
+	switch watchInfo.State {
+	case datapb.ChannelWatchState_Uncomplete, datapb.ChannelWatchState_ToWatch:
```

Comment:
nit: Could use protobuf alias: `option allow_alias = true`(https://developers.google.com/protocol-buffers/docs/proto3)

Reply:
Since we need to deprecated old state, maybe its better to separate `Complete` and `WatchSuccess`.